### PR TITLE
Converts mold mobs to basic mobs

### PR DESCRIPTION
**Pull request by tf-4**
## About The Pull Request

As the title says. Behaviour should be almost entirely preserved, though it's now with the addition of better AI.
Some of this stuff might need some nerfs as a result in the future, but I think it's fine for now.

Also fixes that the oil shamblers would just infinitely stack fire stacks on you but never actually ignite you

## Proof of Testing
<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog
:cl:
refactor: Mold-spawned mobs have been refactored. Expect improved AI behaviours.
fix: Oil shamblers will now actually set you on fire rather than just infinitely stacking fire stacks without igniting them.
/:cl:

### CODE REVIEW

**Review by lessthnthree — APPROVED** [review]

**Comment by RatFromTheJungle** [comment]
@tf-4 add gun mold mob then mayb

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+	gold_core_spawnable = NO_SPAWN
+	faction = list(FACTION_MOLD)
+
+/**
+ * OIL SHAMBLERS
+ *
+ * The mob for the fire mold - humanoids that are (visually) on fire
+ * They're immune to burn damage and have a chance to add fire stacks on melee attack
+ */
+
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+	if(ignite_target.fire_stacks)
+		ignite_target.ignite_mob()
+
+/**
+ * DISEASE MOLD
+ *
+ * Giant rats that are created by the disease mold
+ * They try to inject their target on melee attacks and, if successful, make them contract a disease
+ */
+
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+		to_chat(disease_target, span_danger("[diseased_rat] manages to penetrate your clothing with its teeth!"))
+		disease_target.ForceContractDisease(new diseased_rat.given_disease(), FALSE, TRUE)
+
+/**
+ * ELECTRIC MOLD
+ *
+ * Giant mosquitos spawned by the electric mold
+ * They inject their target with teslium on melee attacks
+ */
+
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+/mob/living/basic/mold/oil_shambler/Initialize(mapload)
+	. = ..()
+	update_overlays()
+
+/mob/living/basic/mold/oil_shambler/update_overlays()
+	. = ..()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+	SSvis_overlays.add_vis_overlay(src, icon, "oil_shambler_overlay", layer, plane, dir, alpha)
+	SSvis_overlays.add_vis_overlay(src, icon, "oil_shambler_overlay", 0, EMISSIVE_PLANE, dir, alpha)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]: Could these strings be defines, and that `0` a define as well?

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+	var/mob/living/carbon/shock_target = target
+	shock_target.reagents.add_reagent(/datum/reagent/teslium, 2)
+
+/**
+ * RADIATION MOLD
+ *
+ * Weird centipede things that spawn with a rad mold
+ * They have a chance to irradiate their target on hit, as well as splashing mutagen on death
+ */
+
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+/mob/living/basic/mold/centaur/Initialize(mapload)
+	. = ..()
+	update_overlays()
+
+/mob/living/basic/mold/centaur/death(gibbed)
+	visible_message(span_warning("[src] ruptures!"))
+	var/datum/reagents/reagent_spawn = new /datum/reagents(300)
+	reagent_spawn.my_atom = src
+	reagent_spawn.add_reagent(death_chem, 20)
+	chem_splash(loc, null, 5, list(reagent_spawn))
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]: These two numbers should be defines.

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+/datum/ai_behavior/basic_melee_attack/centaur/perform(seconds_per_tick, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mold/centaur/centaur = controller.pawn
+
+	if(!isliving(target))
+		return
+	var/mob/living/radiation_target = target
+	if(prob(centaur.irradiate_chance))
+		radiation_pulse(radiation_target, 300, 1, FALSE, TRUE)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]: Same deal here.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Just needs to address the other comments about string defines, and then it would be good to merge.

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mold/oil_shambler/oil_shambler = controller.pawn
+
+	if(!isliving(target))
+		return
+	var/mob/living/ignite_target = target
+	if(prob(oil_shambler.ignite_chance))
+		ignite_target.adjust_fire_stacks(oil_shambler.additional_fire_stacks)
+	if(ignite_target.fire_stacks)
+		ignite_target.ignite_mob()
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
	if(!isliving(target))
		return

	var/mob/living/ignite_target = target
	if(prob(oil_shambler.ignite_chance))
		ignite_target.adjust_fire_stacks(oil_shambler.additional_fire_stacks)

	if(ignite_target.fire_stacks)
		ignite_target.ignite_mob()
```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+/datum/ai_planning_subtree/basic_melee_attack_subtree/diseased_rat
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/diseased_rat
+
+/datum/ai_behavior/basic_melee_attack/diseased_rat/perform(seconds_per_tick, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mold/diseased_rat/diseased_rat = controller.pawn
+
+	if(!isliving(target))
+		return
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
	if(!isliving(target))
		return

```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+
+/datum/ai_planning_subtree/basic_melee_attack_subtree/electric_mosquito
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/electric_mosquito
+
+/datum/ai_behavior/basic_melee_attack/electric_mosquito/perform(seconds_per_tick, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+
+	if(!iscarbon(target))
+		return
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
		return

```

File: modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm [hunk 11]
```diff
@@ -0,0 +1,327 @@
...
+/datum/ai_planning_subtree/basic_melee_attack_subtree/centaur
+	melee_attack_behavior = /datum/ai_behavior/basic_melee_attack/centaur
+
+/datum/ai_behavior/basic_melee_attack/centaur/perform(seconds_per_tick, datum/ai_controller/controller, target_key, targetting_datum_key, hiding_location_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mold/centaur/centaur = controller.pawn
+
+	if(!isliving(target))
+		return
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/biohazard_blob/code/mold_mobs.dm]:
```suggestion
		return

```

**Review by GoldenAlpharex — APPROVED** [review]
Just this to fix and then we good.

**Review by GoldenAlpharex — APPROVED** [review]

**Pull request merged**